astropy/coordinates/funcs.py

```diff
@@ -159,9 +159,9 @@ def get_sun(time):
     # convert barycentric velocity to units of c, but keep as array for passing in to erfa
     earth_v /= c.to(u.au/u.d).value
 
-    dsun = np.sqrt(np.sum(earth_p**2, axis=-1, keepdims=True))
+    dsun = np.sqrt(np.sum(earth_p**2, axis=-1))
     invlorentz = (1-np.sum(earth_v**2, axis=-1))**0.5
```

Comment:
@eteq - I think what was missing was a `keepdims=True` in this line as well. I checked that with that, you do not need the `reshape` anymore.